packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -1,60 +1,35 @@
-import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
 import {
     schema,
     defaultMarkdownParser,
     MarkdownParser
 } from 'prosemirror-markdown';
 import { DOMSerializer } from 'prosemirror-model';
-import { observable } from '@microsoft/fast-element';
-import { template } from './template';
-import { styles } from './styles';
-
-declare global {
-    interface HTMLElementTagNameMap {
-        'nimble-rich-text-viewer': RichTextViewer;
-    }
-}
 
 /**
- * A nimble styled rich text viewer
+ * Provides markdown parser for rich text components
  */
-export class RichTextViewer extends FoundationElement {
-    /**
-     *
-     * @public
-     * Markdown string to render its corresponding rich text content in the component.
-     */
-    @observable
-    public markdown = '';
-
-    /**
-     * @internal
-     */
-    public viewer!: HTMLDivElement;
+export class RichTextMarkdownParser {
```

Comment:
Ideally we'd have unit test coverage for the markdown parser and serializer directly rather than relying on testing in the rich text editor and viewer specs. This might allow us to remove duplication across the component tests. Any thoughts on how feasible it would be to do that?

Reply:
That sounds like a good idea. And I think I'd advocate not doing the folder structure change and the sharing of the model logic in the same PR. I think we should split the PRs so this change is just the folder structure re-organization and we can review de-duplicating the logic and tests in a follow-on PR.

Reply:
Thanks for the feedback. We will address this in the follow-up PR.